extensions/ql-vscode/src/vscode-tests/cli-integration/query.test.ts

```diff
@@ -81,65 +83,36 @@ const queryTestCases: QueryTestCase[] = [
 
 describe('using the query server', function() {
   before(function() {
-    if (process.env['CODEQL_PATH'] === undefined) {
-      console.log('The environment variable CODEQL_PATH is not set. The query server tests, which require the CodeQL CLI, will be skipped.');
+    if (!hasCodeQL()) {
+      console.log('The CodeQL libraries are not available as a folder in this workspace. To fix: checkout the github/codeql repository and set the TEST_CODEQL_PATH envirnment variable to the checked out directory.');
       this.skip();
     }
   });
 
   // Note this does not work with arrow functions as the test case bodies:
   // ensure they are all written with standard anonymous functions.
-  this.timeout(10000);
+  this.timeout(20000);
 
-  const codeQlPath = process.env['CODEQL_PATH']!;
   let qs: qsClient.QueryServerClient;
   let cliServer: cli.CodeQLCliServer;
   const queryServerStarted = new Checkpoint<void>();
-  after(() => {
-    qs?.dispose();
-    cliServer?.dispose();
+
+  beforeEach(async () => {
+    try {
+      const extension = await extensions.getExtension<CodeQLExtensionInterface | {}>('GitHub.vscode-codeql')!.activate();
+      if ('databaseManager' in extension) {
```

Comment:
Maybe check for `cliServer`? It's a bit odd you check for `databaseManager` and then don't use it.